### ISSUE TITLE
PNG with transparency in images manager

### ISSUE THREAD

**Issue by sashabeep** [modxcms/evolution#1132]
Thumbs for PNGs with transparency with black background

![2016-12-21 13 38 08](https://cloud.githubusercontent.com/assets/5689796/21386205/3128fc96-c78b-11e6-9158-644ef1638aaa.png)

**Comment by Deesen** [comment]
This seems to be a server-setting / php-extension issue. For me on some servers BG is correct, on others BG turns black.

**Comment by sashabeep** [comment]
By default only GD is supported in mcpuk, but KCFinder seems to be more customizeable. How to switch from GD to imagick in mcpuk?

**Comment by risingisland** [comment]
Looks like this could probably be used:

`'imageDriversPriority' => "gmagick imagick gd",`

**Comment by sashabeep** [comment]
Tried to add this line to config.php and even changed

```php
/** Default image driver
  * @var string */
    protected $imageDriver = "imagick";
```
in core/uploader.php
but seems like nothing happened. imagick is on

**Comment by bossloper** [comment]
Older version of KCFinder (2.51) works ok. It is the 2.54 version that has introduced a bug.
The problem is here: https://github.com/modxcms/evolution/blob/develop/manager/media/browser/mcpuk/core/uploader.php#L671-#L679

A quick test by changing the return:

`return $img->output("png", array(
            'file' => $thumb,
            'quality' => null
        ));`

and a new png uploaded displays correctly for me. So it is not a graphic driver issue. 
It was apparently fixed in a later version of KCFinder:
https://github.com/sunhater/kcfinder/issues/84

But above the fix cannot simply be applied to 2.54 as the code is quite different.

**Comment by dmi3yy** [comment]
test this: sunhater/kcfinder#84 but its not help 

http://take.ms/y7dVW
http://take.ms/aCwHk

**Comment by bossloper** [comment]
Yes the patch is for something very similar but also a later version so is not a direct patch.

**Comment by janniconl** [comment]
Also discussed before here: https://github.com/modxcms/evolution/issues/663

**Comment by fourroses666** [comment]
So thumbs will be .png in the next version?!
Thats a good solution.

**Comment by dmi3yy** [comment]
Fixed in 1.2.2
